tests/h/search/query_test.py

```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+from __future__ import unicode_literals
+
+import pytest
+
+from h.search import Search
+from h.search import TopLevelAnnotationsFilter
+
+
+class TestTopLevelAnnotationsFilter(object):
+
+    def test_it_filters_out_replies_but_leaves_annotations_in(self, Annotation, search):
+        annotation = Annotation(shared=True)
+        reply = Annotation(references=[annotation.id], shared=True)
+
+        result = search.run({})
+
+        assert annotation.id in result.annotation_ids
+        assert reply.id not in result.annotation_ids
```

Comment:
Yep. Only one simple test is needed. Hopefully man of the filters etc will be this simple to test.